src/pybind/wrapper.cpp

```diff
@@ -186,6 +186,39 @@ except Exception as e:
     return {std::move(solution), std::move(exception_message)};
 }
 
+std::tuple<std::string, std::string> call_diff2c(
+    const std::string& expression,
+    const std::string& variable,
+    const std::unordered_map<std::string, int>& indexed_vars) {
+    std::string statements;
+    // only indexed variables require special treatment
+    for (const auto& [var, prop]: indexed_vars) {
+        statements += fmt::format("_allvars.append(sp.IndexedBase('{}', shape=[1]))\n", var);
```

Comment:
From my testing, it seems the actual value passed to `shape` is not that important since these are not the symbols we are diffing against, but just the "other" vars we need to provide so SymPy handles them properly.